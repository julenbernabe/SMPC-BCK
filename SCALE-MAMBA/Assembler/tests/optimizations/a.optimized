ldi c0, 3
ldsi s0, 3
ldmc c1, 3
ldms s1, 3
stmc c0, 8000
stms s0, 8000
ldmci c2, r0
ldmsi s2, r0
stmci c0, r0
stmsi s0, r0
movc c3, c2
movs s3, s2
movint r1, r0
ldmint r2, 3
stmint r0, 8000
ldminti r3, r2
stminti r3, r2
pushint r3
popint r4
ldtn r1
ldarg r5
reqbl 10
starg r5
run_tape 1, 1, 2
join_tape 3
jmpnz r1, 1
crash
jmpnz r2, 1
restart
clear_memory
clear_registers
addc c6, c1, c2
adds s6, s1, s2
addm s7, s1, c2
addci c7, c1, 10
addsi s8, s1, 10
subc c8, c1, c2
subs s13, s1, s2
subml s9, s13, c1
submr s10, c1, s13
subci c9, c1, 10
subsi s11, s1, 10
subcfi c10, c1, 10
subsfi s10, s1, 10
mulc c11, c1, c2
mulm s11, s1, c2
mulci c12, c1, 10
mulsi s12, s1, 10
divc c13, c3, c2
divci c14, c3, 10
modc c15, c1, c2
modci c16, c1, 10
legendrec c17, c1
digestc c18, c2
output_clear c1, 1
input_clear c19, 1
private_input s13, 1, 10
private_output s13, 2, 10
output_int r6, 10
input_int r7, 10
startopen 4, s1, s2, s3, s4
stopopen 4, c1, c2, c3, c4
opensint r1, sr1
opensbit r1, sb1
triple s1, s2, s3
bit s1
square s1, s2
dabit s1, sb1
ldmsint sr1, 10
ldmsinti sr1, r1
stmsint sr1, 10
stmsinti sr1, r1
movsint sr1, sr2
ldsint sr1, 10
addsint sr1, sr2, sr3
addsintc sr1, sr2, r3
subsint sr1, sr2, sr3
subsintc sr1, sr2, r3
subcints sr1, r2, sr3
mulsint sr1, sr2, sr3
mulsintc sr1, sr2, r3
divsint sr1, sr2, sr3
shlsint sr1, sr2, 10
shrsint sr1, sr2, 10
andc c3, c1, c2
xorc c3, c1, c2
orc c3, c1, c2
andci c3, c1, 10
xorci c3, c1, 10
orci c3, c1, 10
notc c3, c1, 10
neg sr1, sr2
sand sr1, sr2, sb1
xorsb sb3, sb1, sb2
andsb sb3, sb1, sb2
orsb sb3, sb1, sb2
negb sb3, sb1
shlc c3, c1, c2
shrc c3, c1, c2
shlci c3, c1, 10
shrci c3, c1, 10
jmp 1
jmpnz r1, 1
jmpeqz r1, 1
eqzint r2, r1
ltzint r2, r1
ltint r3, r1, r2
gtint r3, r1, r2
eqint r3, r1, r2
ldint r1, 10
addint r1, r2, r3
subint r1, r2, r3
mulint r1, r2, r3
divint r1, r2, r3
convint c1, r2
convmodp r1, c2, 10
convsintsreg sr5, s1
convregsreg sr6, r0
convsregsint s0, sr0
print_mem 10
print_reg c1
print_char 10
print_char4 10
print_char_regint r1
print_char4_regint r1
print_float c1, c2, c3, c4
print_fix c1, 10, 5
print_int r1
eqzsint sb1, sr1
ltzsint sb1, sr1
andsint sr1, sr2, sr3
andsintc sr1, sr2, r6
orsint sr1, sr2, sr3
orsintc sr1, sr2, r6
xorsint sr1, sr2, s3
xorsintc sr1, sr2, r6
invsint sr1, sr2
mul2sint sr1, sr2, sr3, sr4
gc 8
bitsint sb1, sr1, 10
sintbit sr1, sr2, sb6, 10
rand r1, r2
start_clock 10
stop_clock 11
jmp 2
return
call -1
pushint r0 # 3
getspint r8 # 5
peekint r5, r8 # 8
pokeint r8, r2 # 9
popint r4 # 10
pushsint sr0 # 3
getspsint r8 # 5
peeksint sr5, r8 # 8
pokesint r8, sr2 # 9
popsint sr4 # 10
pushs s0 # 3
getsps r8 # 5
peeks s5, r8 # 8
pokes r8, s2 # 9
pops s4 # 10
pushc c0 # 3
getspc r8 # 5
peekc c5, r8 # 8
pokec r8, c2 # 9
popc c4 # 10
pushsbit sb0 # 3
getspsbit r8 # 5
peeksbit sb5, r8 # 8
pokesbit r8, sb2 # 9
popsbit sb4 # 10
ldi c0, 3
