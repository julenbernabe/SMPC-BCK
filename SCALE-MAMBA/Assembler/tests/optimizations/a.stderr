error: unknown or unimplemented instruction: output_share
  --> tests/optimizations/a.asm:57:1
   |
57 | output_share 5, 1, s1, s2, s3, s4
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
error: unknown or unimplemented instruction: input_share
  --> tests/optimizations/a.asm:58:1
   |
58 | input_share 6, 2, s13, s14, s15, s16, s17
   | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
   |
error: unknown or unimplemented instruction: open_chan
  --> tests/optimizations/a.asm:63:1
   |
63 | open_chan r13, 10
   | ^^^^^^^^^^^^^^^^^
   |
error: unknown or unimplemented instruction: close_chan
  --> tests/optimizations/a.asm:64:1
   |
64 | close_chan 10
   | ^^^^^^^^^^^^^
   |
error: unknown or unimplemented instruction: lf_cint
   --> tests/optimizations/a.asm:149:1
    |
149 | lf_cint 21, 42, 5, 4, 3, 2, 1, c11, c12, c13, c14, c15, r1, r2, r3, r4, sr1, sr2, sr3, c1, c2, s1
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
error: unknown or unimplemented instruction: lf_sint
   --> tests/optimizations/a.asm:150:1
    |
150 | lf_sint 21, 42, 5, 4, 3, 2, 1, s11, s12, s13, s14, s15, r1, r2, r3, r4, sr1, sr2, sr3, c1, c2, s1
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
error: unknown or unimplemented instruction: lf_regint
   --> tests/optimizations/a.asm:151:1
    |
151 | lf_regint 21, 42, 5, 4, 3, 2, 1, r11, r12, r13, r14, r15, r1, r2, r3, r4, sr1, sr2, sr3, c1, c2, s1
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
error: unknown or unimplemented instruction: lf_sregint
   --> tests/optimizations/a.asm:152:1
    |
152 | lf_sregint 21, 42, 5, 4, 3, 2, 1, sr11, sr12, sr13, sr14, sr15, r1, r2, r3, r4, sr1, sr2, sr3, c1, c2, s1
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^^
    |
error: unknown or unimplemented instruction: memsizes
   --> tests/optimizations/a.asm:181:1
    |
181 | memsizes 10000
    | ^^^^^^^^^^^^^^
    |
error: unknown or unimplemented instruction: memsizec
   --> tests/optimizations/a.asm:182:1
    |
182 | memsizec 10000
    | ^^^^^^^^^^^^^^
    |
error: unknown or unimplemented instruction: memsizeint
   --> tests/optimizations/a.asm:183:1
    |
183 | memsizeint 10000
    | ^^^^^^^^^^^^^^^^
    |
error: unknown or unimplemented instruction: memsizesint
   --> tests/optimizations/a.asm:184:1
    |
184 | memsizesint 10000
    | ^^^^^^^^^^^^^^^^^
    |
error: expected !do_loop, got `not ssa form`
  --> tests/optimizations/a.asm:20:1
   |
20 | ldtn r1
   | ^^^^^^^ for operand here
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:33:1
   |
33 | adds s6, s1, s2
   | --------------- register 6 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:34:1
   |
34 | addm s7, s1, c2
   | --------------- register 7 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:36:1
   |
36 | addsi s8, s1, 10
   | ---------------- register 8 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:39:1
   |
39 | subml s10, s9, c1
   | ----------------- register 10 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:40:1
   |
40 | submr s11, c1, s9
   | ----------------- register 11 written to but will never be read from again
   |
  ::: tests/optimizations/a.asm:44:1
   |
44 | subsfi s11, s1, 10
   | ------------------ register 11 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:42:1
   |
42 | subsi s12, s1, 10
   | ----------------- register 12 written to but will never be read from again
   |
  ::: tests/optimizations/a.asm:46:1
   |
46 | mulm s12, s1, c2
   | ---------------- register 12 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:48:1
   |
48 | mulsi s13, s1, 10
   | ----------------- register 13 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:32:1
   |
32 | addc c6, c1, c2
   | --------------- register 6 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:35:1
   |
35 | addci c8, c1, 10
   | ---------------- register 8 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:37:1
   |
37 | subc c9, c1, c2
   | --------------- register 9 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:41:1
   |
41 | subci c10, c1, 10
   | ----------------- register 10 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:43:1
   |
43 | subcfi c11, c1, 10
   | ------------------ register 11 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:45:1
   |
45 | mulc c12, c1, c2
   | ---------------- register 12 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:47:1
   |
47 | mulci c13, c1, 10
   | ----------------- register 13 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:49:1
   |
49 | divc c14, c3, c2
   | ---------------- register 14 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:50:1
   |
50 | divci c15, c3, 10
   | ----------------- register 15 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:51:1
   |
51 | modc c16, c1, c2
   | ---------------- register 16 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:52:1
   |
52 | modci c17, c1, 10
   | ----------------- register 17 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:53:1
   |
53 | legendrec c18, c1
   | ----------------- register 18 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:54:1
   |
54 | digestc c19, c2
   | --------------- register 19 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:56:1
   |
56 | input_clear c20, 1
   | ------------------ register 20 written to but will never be read from again
   |
warning: Write to register is never read
  --> tests/optimizations/a.asm:62:1
   |
62 | input_int r11, 10
   | ----------------- register 11 written to but will never be read from again
   |
warning: Write to register is never read
   --> tests/optimizations/a.asm:98:1
    |
 98 | xorsb sb3, sb1, sb2
    | ------------------- register 3 written to but will never be read from again
    |
   ::: tests/optimizations/a.asm:99:1
    |
 99 | andsb sb3, sb1, sb2
    | ------------------- register 3 written to but will never be read from again
    |
   ::: tests/optimizations/a.asm:100:1
    |
100 | orsb sb3, sb1, sb2
    | ------------------ register 3 written to but will never be read from again
    |
   ::: tests/optimizations/a.asm:101:1
    |
101 | negb sb3, sb1
    | ------------- register 3 written to but will never be read from again
    |
warning: Write to register is never read
   --> tests/optimizations/a.asm:122:1
    |
122 | convregsreg sr379, r0
    | --------------------- register 379 written to but will never be read from again
    |
error: read from uninitialized register
   --> tests/optimizations/a.asm:145:1
    |
145 | sintbit sr1, sr2, sb10, 10
    | ^^^^^^^^^^^^^^^^^^^^^^^^^^ register 10 was never written to
    |
warning: Write to register is never read
   --> tests/optimizations/a.asm:168:1
    |
168 | peeks s5, r3000 # 8
    | ------------------- register 5 written to but will never be read from again
    |
warning: Write to register is never read
   --> tests/optimizations/a.asm:173:1
    |
173 | peekc c5, r3000 # 8
    | ------------------- register 5 written to but will never be read from again
    |
error: read from uninitialized register
   --> tests/optimizations/a.asm:176:1
    |
176 | pushsbit sb0 # 3
    | ^^^^^^^^^^^^^^^^ register 0 was never written to
    |
warning: Write to register is never read
   --> tests/optimizations/a.asm:180:1
    |
180 | popsbit sb4 # 10
    | ---------------- register 4 written to but will never be read from again
    |
warning: Write to register is never read
   --> tests/optimizations/a.asm:178:1
    |
178 | peeksbit sb5, r3000 # 8
    | ----------------------- register 5 written to but will never be read from again
    |